### PR TITLE
Add config params

### PR DESCRIPTION
**Pull request by teadur**
Add some flexibility to configuration, shouldnot brake anything for users using the role

### CODE REVIEW

File: tasks/install.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@
   group:
     name: ssmtp
     state: present
+  when: "{{ ssmtp.restrict == true }}"
```
Comment (lobsterdore) [inline tasks/install.yml]: When clauses should be bare as far as I know eg.

```YAML
when: ssmtp.restrict == true
```

In fact you don't need the equality test here:

```YAML
when: ssmtp.restrict
```

File: templates/ssmtp.conf.j2 [hunk 2]
```diff
@@ -1,7 +1,20 @@
 mailhub={{ ssmtp.config.host }}:{{ ssmtp.config.port }}
 
+{% if ssmtp.config.root is defined %}
```
Comment (lobsterdore) [inline templates/ssmtp.conf.j2]: Can you define these settings (ssmtp.config.root, ssmtp.config.rewritedomain and the like) but set them to false in defaults? I prefer to have all possible values defined in defaults so it's obvious what options are available.

File: defaults/main.yml [hunk 3]
```diff
@@ -4,7 +4,10 @@ ssmtp:
   # User and group that will be able to use ssmtp
   group: ~
   user: ~
-
+  restrict: true
```
Comment (lobsterdore) [inline defaults/main.yml]: Can you alphabetically order these vars? So it should be:

```YAML
ssmtp:
  auth: true
  # User and group that will be able to use ssmtp
  group: ~
  restrict: true
  revaliases: true
  use_inventory_hostname: false
  user: ~
```

**Comment by lobsterdore** [comment]
When your changes are finished please squash into one meaningful commit at well.

File: tasks/install.yml (hunk 1 above)
Comment (teadur) [reply tasks/install.yml]: indeed, thanks for showing me how todo it correctly

File: templates/ssmtp.conf.j2 (hunk 2 above)
Comment (teadur) [reply templates/ssmtp.conf.j2]: Should be okey now

File: defaults/main.yml (hunk 3 above)
Comment (teadur) [reply defaults/main.yml]: Should be okey now

**Comment by teadur** [comment]
Do i understand correctly that after squash i should make new pull request ? 
If everything is okey with you i shall do it.
For squash is comment " Allow configuration of hostname via inventory_hostname,
configuration of auth, rewritedomain, revaliases, root, fromlineoverride and ssmtp restriction." descriptive enough ?

**Comment by lobsterdore** [comment]
@teadur no need for a new pull request, just squash your commits in your branch and that will be fine. That commit message sounds fine as well.

**Comment by lobsterdore** [comment]
LGTM 👍

**Pull request merged**